src/saturn_engine/worker/topics/logger.py

```diff
@@ -0,0 +1,71 @@
+from typing import AsyncGenerator
+from typing import cast
+
+import dataclasses
+import logging
+from enum import Enum
+
+from saturn_engine.core import TopicMessage
+from saturn_engine.core.error import ErrorMessageArgs
+from saturn_engine.worker.topic import Topic
+
+
+class Level(Enum):
+    IGNORE = None
+    DEBUG = logging.DEBUG
+    INFO = logging.INFO
+    WARNING = logging.WARNING
+    ERROR = logging.ERROR
+
+
+LevelMap: dict[str, Level] = {
+    "ignore": Level.IGNORE,
+    "debug": Level.DEBUG,
+    "info": Level.INFO,
+    "warning": Level.WARNING,
+    "error": Level.ERROR,
+}
+
+
+class LoggingTopic(Topic):
+    """A topic contains default loggers for error handling
+    such as ignoring the error and logging it in debug.
+    """
+
+    @dataclasses.dataclass
+    class Options:
+        name: str
+        level: str
+
+    def __init__(self, options: Options, **kwargs: object) -> None:
+        self.level = LevelMap[options.level]
+        self.name = options.name
+        self.logger = logging.getLogger(f"saturn.topics.{self.name}")
+
+    async def run(self) -> AsyncGenerator[TopicMessage, None]:
+        return
+        yield
+
+    async def publish(self, message: TopicMessage, wait: bool) -> bool:
+        if self.level == Level.IGNORE:
+            return True
+
+        extra = {"data": {"id": message.id, "args": message.args}}
+        msg = f"Message {message.id} executed successfully"
+
+        if "error" in message.args:
+            # remove args from extra since it would be redundant with the traceback
+            del extra["data"]["args"]
+            # get our error message and put the traceback in our extra data
+            error_msg: ErrorMessageArgs = cast(ErrorMessageArgs, message.args["error"])
```

Comment:
don't cast, `isinstance` instead. You might want to do the `isinstance` before the `del`, to make sure we really are dealing with an `ErrorMessageArgs` and not some other `error` arg. aka

```python
if isinstance(message.args.get("error"), ErrorMessageArgs):
  # do stuff here
```